cpp/include/cudf/detail/utilities/hash_functions.cuh

```diff
@@ -187,10 +121,97 @@ struct MurmurHash3_32
         h1 = fmix32(h1);
         return h1;
     }
+
 private:
     uint32_t m_seed;
 };
 
+template <>
+hash_value_type CUDA_HOST_DEVICE_CALLABLE MurmurHash3_32<cudf::bool8>::operator()(cudf::bool8 const& key) const {
+  return this->compute(static_cast<int8_t>(key));
+}
+
+template <>
+hash_value_type CUDA_HOST_DEVICE_CALLABLE MurmurHash3_32<cudf::experimental::bool8>::operator()(cudf::experimental::bool8 const& key) const {
+  return this->compute(static_cast<uint8_t>(key));
+}
+
+/**
+ * @brief Specialization of MurmurHash3_32 operator for strings.
+ */
+template <>
+hash_value_type CUDA_HOST_DEVICE_CALLABLE MurmurHash3_32<cudf::string_view>::operator()(cudf::string_view const& key) const {
+  const int len = (int)key.size_bytes();
+  const uint8_t* data = (const uint8_t*)key.data();
+  const int nblocks = len / 4;
+  result_type h1 = m_seed;
+  constexpr uint32_t c1 = 0xcc9e2d51;
+  constexpr uint32_t c2 = 0x1b873593;
+  auto getblock32 = [] __host__ __device__(const uint32_t* p, int i) -> uint32_t {
+    // Individual byte reads for unaligned accesses (very likely)
+#ifndef __CUDA_ARCH__
+    CUDF_FAIL("Hashing a string in host code is not supported.");
+#else
+    auto q = (const uint8_t*)(p + i);
+    return q[0] | (q[1] << 8) | (q[2] << 16) | (q[3] << 24);
+#endif
+  };
+
+  //----------
+  // body
+  const uint32_t* const blocks = (const uint32_t*)(data + nblocks * 4);
+  for (int i = -nblocks; i; i++) {
+    uint32_t k1 = getblock32(blocks, i);
+    k1 *= c1;
+    k1 = rotl32(k1, 15);
+    k1 *= c2;
+    h1 ^= k1;
+    h1 = rotl32(h1, 13);
+    h1 = h1 * 5 + 0xe6546b64;
+  }
+  //----------
+  // tail
+  const uint8_t* tail = (const uint8_t*)(data + nblocks * 4);
+  uint32_t k1 = 0;
+  switch (len & 3) {
+  case 3: k1 ^= tail[2] << 16;
+  case 2: k1 ^= tail[1] << 8;
+  case 1: k1 ^= tail[0];
+    k1 *= c1; k1 = rotl32(k1, 15); k1 *= c2; h1 ^= k1;
+  };
+  //----------
+  // finalization
+  h1 ^= len;
+  h1 = fmix32(h1);
+  return h1;
+}
+
+// Specialization for float, make sure 0.0 and NaN hash consistently
+template <>
+hash_value_type CUDA_HOST_DEVICE_CALLABLE MurmurHash3_32<float>::operator()(float const& key) const {
+  if (key == 0.0f) {
+    return 0;
+  } else if (isnan(key)) {
+    float nan = std::numeric_limits<float>::quiet_NaN();
+    return this->compute(nan);
+  } else {
+    return this->compute(key);
+  }
```

Comment:
Same implementation for double? Can be extracted into a new templated function to reduce duplication.